src/components/autocomplete/js/autocompleteDirective.js

```diff
@@ -165,19 +170,17 @@ function MdAutocomplete ($mdTheming, $mdUtil) {
         </aria-status>';
 
       function getItemTemplate() {
-        var templateTag = element.find('md-item-template').remove(),
+        var templateTag = element.find('md-item-template').detach(),
             html = templateTag.length ? templateTag.html() : element.html();
         if (!templateTag.length) element.empty();
         return html;
       }
 
       function getNoItemsTemplate() {
-        var templateTag = element.find('md-not-found').remove(),
+        var templateTag = element.find('md-not-found').detach(),
```

Comment:
https://github.com/angular/angular.js/blob/master/src/jqLite.js

remove will call into jqLiteDealloc, which tries to find data associated with the elements to clean up. In this case we can be confident that such data does not exist, so detach is faster.